### PR TITLE
Docs for Pricing

### PR DESCRIPTION
**Pull request by andyzg**
image
image
image
image
image
image
image
image
image
image
image

### CODE REVIEW

File: windsurf/plan-usage.mdx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+If a tool call errors or a user message is unsuccessful, flow action and User Prompt credits will not be consumed. For example, if Cascade attempts to write to a file but that file has unsaved changes, the tool call will fail and it will not consume a credit.
+
+### Upgrading to a paid plan
+
+To learn more about paid features or to upgrade to a paid plan, [click here](https://codeium.com/plan). Paid plans include Pro and Pro Ultimate options for individuals as well as for teams.
+
+### Purchasing additional Flex credits
+
+If you run out of user prompt or Flow Action credits, you can purchase Flex credits in the [billing website](https://codeium.com/plan). Flex credits are a credit that can be used as both User Prompt and Flow Action credits. For example, if you run out of tool calls but still have remaining User Prompt credits, your Flex credits will be consumed as Flow Action credits. If you run out of both User Prompt and Flow Action credits, Flex credits will be consumed as both. In other words, if you send 1 premium model message that performs 4 tool calls, you'll consume 1 user prompt + 4 flow actions = 5 Flex credits.
```
Comment (aunch) [inline windsurf/plan-usage.mdx:39]: nit: let's make sure we keep things like `User Prompt` capitalized and use `User Prompt credit` and `Flow Action credit`

File: windsurf/plan-usage.mdx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+If you're on a Teams plan, contact your Teams admin to purchase more Flex credits to increase your amount of Flex credits. For more on the cost of Flex credits, visit our [pricing page](LINK)
```
Comment (aunch) [inline windsurf/plan-usage.mdx:41]: nit: Link

File: windsurf/plan-usage.mdx (hunk 1 above)
Comment (aunch) [reply windsurf/plan-usage.mdx:39]: last sentence suggestion: 

For example, if you are out of both User Prompt credits and Flow Action credits, send a message to Cascade with a premium model, and it performs tool calls as part of the response, you would consume 5 Flex credits, the equivalent of 1 User Prompt and 4 Flow Action credits.

File: windsurf/plan-usage.mdx (hunk 2 above)
Comment (aunch) [reply windsurf/plan-usage.mdx:41]: probably mention pooling for Teams?

For Team plans, Flex credits are purchased within and treated as a pool amongst all members of the team.

File: windsurf/plan-usage.mdx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+## Usage examples
+
+To explain how credits work, here are a few examples.
+
+In this first example below, the user has both User Prompt and Flow Action credits available. In the scenario illustrated, 1 user action credit was consumed and 4 Flow Action credits were consumed.
```
Comment (aunch) [inline windsurf/plan-usage.mdx:47]: same nit to keep naming and capitalization consistent (instead of user action credit, use User Prompt credit), will stop commenting throughout but probably good to take a closer pass because i see at least a dozen such instances in the rest

Comment (aunch) [reply windsurf/plan-usage.mdx:47]: user prompt vs user action will definitely cause confusion

File: windsurf/plan-usage.mdx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+</Frame>
+
+In this final example below, the user has no user prompt or Flow Action credits available but they've purchased Flex credits. Since Flex credits can be used for both User Prompt and Flow Action credits, they both count as Flex credits. In the scenario illustrated, 5 Flex credits were consumed.
+
+<Frame>
+  <img src="/assets/windsurf/plan-info/credit-example-only-Flex.png" />
+</Frame>
+
+If there are no Flex credits available and one of user prompt or Flow Action credits are unavailable,
+then Write and Chat mode will be disabled with premium models. Cascade Base model will still be available for user in Write and Chat mode.
```
Comment (aunch) [inline windsurf/plan-usage.mdx:72]: might be good to copy in answers for what to do when one of User Prompt/Flow Action runs out from https://docs.google.com/document/d/1AhVmnWFKL9VA8GvT5lbgRu2dnTB6RaEKfVxzWYRlPjY/edit?tab=t.0?

in fact probably a number of those FAQs answers are good. sorry for updating them so late...

File: windsurf/plan-usage.mdx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+### Using Pro Ultimate Plan
+
+While on Pro Ultimate, you'll have access to a monthly quota of Flow Action credits and an **unlimited number of User Prompt credits**. Tool calls within Cascade will consume your Flow Action credits. You can view how many credits you have remaining in the Windsurf Settings panel that's accessible in the status bar.
+
+If you're running low on Flow Action credits, Cascade will notify you so that you can purchase additional Flex credits. To purchase additional Flex credits, visit the billing website and select “Purchase flex credits”. The Flex credits purchased will rollover to the following usage month if there are any remaining.
+
+If you want to reduce your usage of Flow Action credits, you have two options:
+
+- Use Legacy Chat mode with a premium model. On Pro Ultimate, you have unlimited prompt credits so Legacy Chat usage does not consume any credits.
```
Comment (aunch) [inline windsurf/plan-usage.mdx:109]: Legacy mode, not Legacy Chat mode

File: windsurf/plan-usage.mdx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+- Use Cascade Base model in Write or Chat mode. Using Cascade Base model also does not consume any credits.
+
+In addition to increased Cascade credits, Pro Ultimate comes with unlimited Fast Autocomplete and unlimited premium model requests with Command.
+
+If you'd like to downgrade to Pro, visit the billing website. Upon downgrading, you'll remain in Pro Ultimate until the end of your billing cycle. At the end of your billing cycle, your plan will change to Pro. Any Flex credits will transfer over.
+
+### Using a Free plan
+
+If you're not ready to upgrade to a paid plan, Windsurf can still be utilized on the Free plan! While in Cascade, you'll be given limited number of User Prompt and. Flow Action credits and have access to our Cascade Base model in read mode to ask questions. When editing code, you'll have access to unlimited autocomplete suggestions and AI command instructions. To learn more about features in Free and in paid tiers, [click here](https://codeium.com/pricing).
```
Comment (aunch) [inline windsurf/plan-usage.mdx:118]: grammar/punctuation of this first sentence is off?

File: windsurf/plan-usage.mdx [hunk 7]
```diff
@@ -0,0 +1,122 @@
+---
+title: "Paid Plan and Credit Usage"
+description: ""
+---
+
+## Getting started
+
```
Comment (aunch) [inline windsurf/plan-usage.mdx:7]: would reword last sentence:

One premium User Prompt credit is consumed whenever a message is sent to Cascade with a premium model (e.g. GPT-4o, Sonnet) while one premium Flow Action credit is consumed when a premium model is used by the AI in Write and Chat mode for a tool call (e.g. search, analyze, write, terminal command, etc.).

**Review by khou22 — APPROVED** [review]

**Review by aunch — COMMENTED** [review]
quick pass for things that stood out

**Review by roberthou1 — APPROVED** [review]
<sub>_[Reviewed with Forge](https://codeium.com/forge/Exafunction/codeium-docs/42/review)_</sub>
